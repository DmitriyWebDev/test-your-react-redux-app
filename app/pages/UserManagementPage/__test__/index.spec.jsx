import React from 'react';
import { shallow } from 'enzyme';
import { UserManagementPage } from '../index';
import expect from 'expect';
import sinon from 'sinon';

describe('UserManagementPage component', () => {
  const defaultProps = {
    usersList: [],
    pageIsLoading: false,
    userAdditionIsLoaded: false,
    userDeletingIsLoaded: false,
    getUsersList: () => ({}),
    addUser: () => ({}),
    deleteUser: () => ({}),
    resetPageData: () => ({}),
  };

  it('should render correctly', () => {
    const component = shallow(<UserManagementPage {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  /** Test events handling (in react components lifecycle hooks) */

  it('Requests users on lifecycle hook "ComponentDidMount"', () => {
    const getUsersList = jest.fn();
    const props = {
      ...defaultProps,
      getUsersList,
    };

    shallow(<UserManagementPage {...props} />);

    expect(getUsersList).toBeCalledTimes(1);
  });

  it('Requests users on lifecycle hook "ComponentDidUpdate", if user added', () => {
    const wrapper = shallow(<UserManagementPage {...defaultProps} />);
    const props = {
      ...defaultProps,
      getUsersList: () => ({}),
      userAdditionIsLoaded: false,
    };
    const getUsersList = sinon.stub(props, 'getUsersList');

    wrapper.setProps({ ...props, userAdditionIsLoaded: true });

    expect(getUsersList.calledOnce).toBe(true);
  });
});
