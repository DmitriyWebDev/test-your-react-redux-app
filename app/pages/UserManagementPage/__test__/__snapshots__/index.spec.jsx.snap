// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserManagementPage component should render correctly 1`] = `
<div
  className="userManagementPage"
>
  <div
    className="userManagementPage__contentWrap"
  >
    <div
      className="userManagementPage__header"
    >
      User management page
    </div>
    <div
      className="userManagementPage__content"
    >
      <div
        className="userManagementPage__section"
      >
        <div
          className="userManagementPage__sectionContent"
        >
          <UsersList
            deleteUser={[Function]}
            disabled={false}
            users={Array []}
          />
        </div>
      </div>
      <div
        className="userManagementPage__section"
      >
        <div
          className="userManagementPage__sectionContent"
        >
          <FormAddUser
            addUser={[Function]}
            disabled={false}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
